lib/train-winrm/connection.rb

```diff
@@ -189,10 +189,10 @@ def login_command_for_windows
       # @api private
       def session(retry_options = {})
         @session ||= begin
-          opts = {
+          opts = retry_options.merge({
             retry_limit: @connection_retries.to_i,
             retry_delay: @connection_retry_sleep.to_i,
-          }.merge(retry_options)
+          })
```

Comment:
This doesn't seem right to me. The usual order of preference overrides is: global -> ENV -> cmdline -> object -> method call... So that big loses to little. This change means that the options passed to the local `session` call are overridden by the object's ivars.

Also, a change like this really should have a test confirming the problem ad the fix.